plugins/integrations/kubernetes-service/src/main/resources/conf/k8s-master.yml

```diff
@@ -0,0 +1,265 @@
+        CNI_VERSION="v0.7.5"
+        CRICTL_VERSION="v1.16.0"
+        echo "Warning: ${BINARIES_DIR} not found. Will get binaries and docker images from Internet."
+        mkdir -p /opt/cni/bin
+        curl -L "https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-amd64-${CNI_VERSION}.tgz" | tar -C /opt/cni/bin -xz
+
+        mkdir -p /opt/bin
+        curl -L "https://github.com/kubernetes-incubator/cri-tools/releases/download/${CRICTL_VERSION}/crictl-${CRICTL_VERSION}-linux-amd64.tar.gz" | tar -C /opt/bin -xz
+
+        mkdir -p /opt/bin
+        cd /opt/bin
+        curl -L --remote-name-all https://storage.googleapis.com/kubernetes-release/release/${RELEASE}/bin/linux/amd64/{kubeadm,kubelet,kubectl}
+        chmod +x {kubeadm,kubelet,kubectl}
+
+        curl -sSL "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/kubelet.service" | sed "s:/usr/bin:/opt/bin:g" > /etc/systemd/system/kubelet.service
+        mkdir -p /etc/systemd/system/kubelet.service.d
+        curl -sSL "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/10-kubeadm.conf" | sed "s:/usr/bin:/opt/bin:g" > /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+      fi
+
+      systemctl enable kubelet && systemctl start kubelet
+      modprobe br_netfilter && sysctl net.bridge.bridge-nf-call-iptables=1
+
+      if [ -d "$BINARIES_DIR" ] && [ "$ATTEMPT_ONLINE_INSTALL" = true ]; then
+        crucial_cmd_attempts=1
+        while true; do
+          if (( "$crucial_cmd_attempts" > "$MAX_SETUP_CRUCIAL_CMD_ATTEMPTS" )); then
+            echo "Warning: kubeadm pull images failed after multiple tries!"
+            break;
+          fi
+          retval=0
+          set +e
+          kubeadm config images pull
+          retval=$?
+          set -e
+          if [ $retval -eq 0 ]; then
+            break;
+          fi
+          crucial_cmd_attempts=$[$crucial_cmd_attempts + 1]
+        done
+      fi
+
+      crucial_cmd_attempts=1
+      while true; do
+        if (( "$crucial_cmd_attempts" > "$MAX_SETUP_CRUCIAL_CMD_ATTEMPTS" )); then
+          echo "Error: kubeadm init failed!"
+          exit 1
+        fi
+        retval=0
+        set +e
+        kubeadm init --token {{ k8s_master.cluster.token }} {{ k8s_master.cluster.initargs }}
+        retval=$?
+        set -e
+        if [ $retval -eq 0 ]; then
+          break;
+        fi
+        crucial_cmd_attempts=$[$crucial_cmd_attempts + 1]
+      done
+
+  - path: /opt/bin/deploy-kube-system
+    permissions: 0700
+    owner: root:root
+    content: |
+      #!/bin/bash -e
+
+      if [[ -f "/home/core/success" ]]; then
+      echo "Already provisioned!"
+      exit 0
+      fi
+
+      ISO_MOUNT_DIR=/mnt/k8sdisk
+      BINARIES_DIR=${ISO_MOUNT_DIR}/
+
+      if [[ $(systemctl is-active setup-kube-system) != "inactive" ]]; then
+        echo "setup-kube-system is running!"
+        exit 1
+      fi
+      export PATH=$PATH:/opt/bin
```

Comment:
The PATH gets really long here since the deploy is waiting setup to finish which takes minute(s). In my VM, the path was like:

```bash
w-1-k8s-master ~ # echo $PATH
/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:/opt/bin:
```

Reply:
@poussa tried to fix this with latest commit

Reply:
Actually, this only happens if the deploy-kube-system fails for some reason and systemd restarts it.

Reply:
@poussa Added changes should take of that as well